sdks/python/apache_beam/ml/inference/huggingface_inference.py

```diff
@@ -0,0 +1,462 @@
+from transformers import AutoModel
+from transformers import TFAutoModel
+
+__all__ = [
+    'HuggingFaceModelHandlerTensor',
+    'HuggingFaceModelHandlerKeyedTensor',
+]
+
+TensorInferenceFn = Callable[[
+    Sequence[Union[torch.Tensor, tf.Tensor]],
+    Union[AutoModel, TFAutoModel],
+    torch.device,
+    Optional[Dict[str, Any]],
+    Optional[str]
+],
+                             Iterable[PredictionResult]]
+
+KeyedTensorInferenceFn = Callable[[
+    Sequence[Dict[str, Union[torch.Tensor, tf.Tensor]]],
+    Union[AutoModel, TFAutoModel],
+    torch.device,
+    Optional[Dict[str, Any]],
+    Optional[str]
+],
+                                  Iterable[PredictionResult]]
+
+
+def _validate_constructor_args(model_uri, model_class):
+  message = (
+      "Please provide both model class and model uri to load the model."
+      "Got params as model_uri={model_uri} and "
+      "model_class={model_class}.")
+  if not model_uri and not model_class:
+    raise RuntimeError(
+        message.format(model_uri=model_uri, model_class=model_class))
+  elif not model_uri:
+    raise RuntimeError(
+        message.format(model_uri=model_uri, model_class=model_class))
+  elif not model_class:
+    raise RuntimeError(
+        message.format(model_uri=model_uri, model_class=model_class))
+
+
+def _run_inference_torch_keyed_tensor(
+    batch: Sequence[Dict[str, Union[tf.Tensor, torch.Tensor]]],
+    model: AutoModel,
+    device,
+    inference_args: Dict[str, Any],
+    model_id: Optional[str] = None) -> Iterable[PredictionResult]:
+  key_to_tensor_list = defaultdict(list)
+  # torch.no_grad() mitigates GPU memory issues
+  # https://github.com/apache/beam/issues/22811
+  with torch.no_grad():
+    for example in batch:
+      for key, tensor in example.items():
+        key_to_tensor_list[key].append(tensor)
+    key_to_batched_tensors = {}
+    for key in key_to_tensor_list:
+      batched_tensors = torch.stack(key_to_tensor_list[key])
+      batched_tensors = _convert_to_device(batched_tensors, device)
+      key_to_batched_tensors[key] = batched_tensors
+    return utils._convert_to_result(
+        batch, model(**key_to_batched_tensors, **inference_args))
+
+
+def _run_inference_tensorflow_keyed_tensor(
+    batch: Sequence[Dict[str, Union[tf.Tensor, torch.Tensor]]],
+    model: TFAutoModel,
+    device,
+    inference_args: Dict[str, Any],
+    model_id: Optional[str] = None) -> Iterable[PredictionResult]:
+  key_to_tensor_list = defaultdict(list)
+  for example in batch:
+    for key, tensor in example.items():
+      key_to_tensor_list[key].append(tensor)
+  key_to_batched_tensors = {}
+  for key in key_to_tensor_list:
+    batched_tensors = torch.stack(key_to_tensor_list[key])
+    batched_tensors = key_to_tensor_list[key]
+    key_to_batched_tensors[key] = batched_tensors
+  return utils._convert_to_result(
+      batch, model(**key_to_batched_tensors, **inference_args))
+
+
+class HuggingFaceModelHandlerKeyedTensor(ModelHandler[Dict[str,
```

Comment:
Can we follow a pattern where we create a class `HuggingFaceModelHandler`, which would have common method across `HuggingFaceModelHandlerTensor/KeyedTensor` such as `load_model` so that we wouldn't edit the twice? For the types we can use generic types for `HuggingFaceModelHandler`

XGBoostModelHandler was implemented this way https://github.com/apache/beam/blob/5f1eae622932bc3362731bcc8cf464bf678877d4/sdks/python/apache_beam/ml/inference/xgboost_inference.py#L185